Source/WebCore/svg/SVGURIReference.h

```diff
@@ -54,7 +54,6 @@ class SVGURIReference {
 
         // If the URI matches our documents URL, we're dealing with a local reference.
         URL url = document.completeURL(uri);
-        ASSERT(!url.protocolIsData());
```

Comment:
What is the goal of this change?

Removing this assertion will prevent a crash in the debug build. But it will not affect the release build. More importantly this change will not fix the actual bug which can happen in cases like imported/w3c/web-platform-tests/css/filter-effects/svg-feimage-002.html.

The code in `SVGURIReference::targetElementFromIRIString()` assumes the `targetElement` is either local in the current `treeScope` or it is an element inside an `externalDocument`. It does not handle the case when the URL is data url followed by the fragment ID. So we end up making this early return:
```
    // Exit early if the referenced url is external, and we have no externalDocument given.
    if (isExternalURIReference(iri, document))
        return { nullptr, WTFMove(id) };
```
I think fixing the actual bug is more important than removing an assertion which reminds us that our code is not complete yet.

Reply:
Updated but FE test is not fixed yet. I wonder if we could file an different bug on that. PTAL. Thank you!